syntax = "proto3";

package smartclass;

service SmartClass {
  rpc JoinRoom (JoinRoomRequest) returns (JoinRoomResponse);
  rpc GetNextQuestion (Player) returns (Question);
  rpc SubmitAnswer (Answer) returns (GameStatus);
  rpc ExitRoom (Player) returns (GameStatus);
}

message JoinRoomRequest {
  string player_name = 1;
}

message JoinRoomResponse {
  bool joined = 1;
  int32 game_timer = 2;
  string message = 4;
}

message Player {
  string player_name = 1;
}

message Question {
  string question = 1;
  repeated string options = 2;
  string answer = 3;
}

message Answer {
  string player_name = 1;
  bool isCorrect = 2;
}

message GameStatus {
  bool isEnd = 1;
  int32 score = 2;
}